src/all/ReVector.java

```
@@ -24,16 +18,16 @@
 	}
 
 	//Add object _o to the end of Vector
-	void add(T _o)
+	public void add(T _o)
 	{
 		Object temp[] = _array.clone();
 		_array =  new Object[_array.length + 1];
 		System.arraycopy(temp, 0, _array, 0, temp.length);
 		add(_array.length-1, _o);
 	}
-
+	
 	//Add object _0 by index to a Vector
-	void add(int _index, T _o)
+	public void add(int _index, T _o)
 	{
 		if(_index < _array.length)
 			_array[_index] = _o;
```

Don't do some extra work. You should throw an exception if index > size.